Sources/XCTQueues/TestQueueDriver.swift

```diff
@@ -0,0 +1,83 @@
+import Queues
+import Vapor
+
+extension Application.Queues.Provider {
+    public static var test: Self {
+        .init {
+            $0.queues.use(custom: TestQueuesDriver())
+        }
+    }
+}
+
+public struct TestQueuesDriver: QueuesDriver {
+    public func makeQueue(with context: QueueContext) -> Queue {
+        TestQueue(context: context)
+    }
+    
+    public func shutdown() {
+        // nothing
+    }
+}
+
+extension Application.Queues {
+    public final class TestQueueStorage {
+        public var jobs: [JobIdentifier: JobData] = [:]
+        public var queue: [JobIdentifier] = []
```

Comment:
I wonder what the best way to expose these properties is. `JobData`'s payload is just an array of bytes which isn't particularly useful. I wonder if we could do something like:

```swift
func first<J>(_ job: J.Type) -> J.Payload?
    where J: Job
```

That way the user can assert values from the actual job payload which I think is what they will care about.